pkg/expression/builtin_json.go

```diff
@@ -1796,3 +1800,72 @@ func (b *builtinJSONLengthSig) evalInt(ctx EvalContext, row chunk.Row) (res int6
 	}
 	return int64(obj.GetElemCount()), false, nil
 }
+
+type jsonSchemaValidFunctionClass struct {
+	baseFunctionClass
+}
+
+func (c *jsonSchemaValidFunctionClass) getFunction(ctx BuildContext, args []Expression) (builtinFunc, error) {
+	if err := c.verifyArgs(args); err != nil {
+		return nil, err
+	}
+	bf, err := newBaseBuiltinFuncWithTp(ctx, c.funcName, args, types.ETInt, types.ETJson, types.ETJson)
+	if err != nil {
+		return nil, err
+	}
+
+	sig := &builtinJSONSchemaValidSig{bf}
+	return sig, nil
+}
+
+type builtinJSONSchemaValidSig struct {
+	baseBuiltinFunc
+}
+
+func (b *builtinJSONSchemaValidSig) Clone() builtinFunc {
+	newSig := &builtinJSONSchemaValidSig{}
+	newSig.cloneFrom(&b.baseBuiltinFunc)
+	return newSig
+}
+
+func (b *builtinJSONSchemaValidSig) evalInt(ctx EvalContext, row chunk.Row) (res int64, isNull bool, err error) {
+	schema := &jsonschema.Schema{}
+
+	// First argument is the schema
+	schemaData, schemaIsNull, err := b.args[0].EvalJSON(ctx, row)
```

Comment:
Considering that for most user cases, the `schema` should be a constant, how about adding some optimazations for the case when `schema` is a constant? An example can be found in the `like` function:
https://github.com/pingcap/tidb/blob/v8.0.0/pkg/expression/builtin_like.go#L56

Reply:
I've done this in c5c2feb06dfe2a6e13d9d7f3d7942bac7cd61901